google-http-client/src/main/java/com/google/api/client/util/store/FileDataStoreFactory.java

```diff
@@ -53,7 +53,7 @@
   private static final Logger LOGGER = Logger.getLogger(FileDataStoreFactory.class.getName());
 
   private static final boolean IS_WINDOWS = StandardSystemProperty.OS_NAME.value()
-      .startsWith("WINDOWS");
+      .regionMatches(true, 0, "WINDOWS", 0, "WINDOWS".length());
```

Comment:
This appears to depend on the default character set.

Reply:
We are just dealing with strings here, they are always in utf-16. Do you mean default locale? If so, I don't think so. The Javadoc for `regionMatches` does not mention anything about locale and the source code also does not do anything locale dependent.

Reply:
As for testing...
The PR that introduced this check mentions waiting for a windows test machine, which was then apparently addressed in a different PR. I am not familiar with the automated testing setup for this library, but through searching the web I have not been able to find an instance where `os.name` _ever_ contained `"WINDOWS"` (all uppercase). So I am not sure how the initial PR even passed testing on any kind of Windows system.

Reply:
See: https://github.com/googleapis/google-http-java-client/pull/557

Reply:
Case insensitive comparison is fundamentally locale dependent. E.g. I is not the upper case of i in Turkish locales. UTF-16 is a character set, not a locale.

Reply:
> UTF-16 is a character set, not a locale.
Yes. Which is why your original message saying "default character set" confused me.

> Case insensitive comparison is fundamentally locale dependent. E.g. I is not the upper case of i in Turkish locales. 
I know. `regionMatches` does not take this into account. `regionMatches` uses `Character.toLowerCase`, which specifically mentions:
> {@code String} case mapping methods can perform locale-sensitive mappings, context-sensitive mappings, and 1:M character mappings, whereas the {@code Character} case mapping methods cannot.

If you prefer i'll change it to a `toLowerCase(Locale.ENGLISH)` version though.

Reply:
Yes, that would be good. I still need to think about how to test this.